docs/nodejs_dev_guide.rst

```diff
@@ -215,148 +199,359 @@ Add app.js and update package.json's scripts
 
-Run on dev mode
----------------
 
-Open a new terminal window, `ssh` into the ``teracy-dev`` VM to execute the following commands:
+Run on prod mode
+----------------
+
+Open a new terminal window,, `ssh` into the ``teracy-dev`` VM to execute the following commands:
 
 ..  code-block:: bash
 
     $ vagrant ssh
     $ ws
-    $ cd hello-world-nodejs
-    $ docker-compose up -d
+    $ cd nodejs-hello-world
+    $ docker-compose -f docker-compose.prod.yml build prod # to build the prodution Docker image
+    $ docker-compose -f docker-compose.yml -f docker-compose.prod.yml up -d prod && \
+      docker-compose -f docker-compose.yml -f docker-compose.prod.yml logs -f prod # to run the production Docker image
 
 
-After that, open hello-d.teracy.dev on your browser to see the app on the dev mode.
+After that, open:
 
-Run on prod mode
-----------------
+- http://nodejs.teracy.dev or https://nodejs.teracy.dev on your host browser to see the app
+  on the dev mode.
+- Check out the VM's :ref:`basic_usage-ip_address` and on any device within your LAN,
+  open http://nodejs.<vm_ip>.xip.io or https://nodejs.<vm_ip>.xip.io to see the web app.
+
+
+Create review mode
+------------------
+
+Review mode will be used to review the production docker image from other team members.
+
+This is very simple docker-compose configuration.
+
+The changes should be like this: https://github.com/acme101/nodejs-hello-world/commit/1f7d0cfdd82435a482834fb45b2203674134cfc2
+
+
+Run on review mode
+------------------
+
+For example, we're going to review `hoatle/nodejs-hello-world:feature-1` Docker image.
 
 Open a new terminal window,, `ssh` into the ``teracy-dev`` VM to execute the following commands:
 
 ..  code-block:: bash
 
     $ vagrant ssh
     $ ws
-    $ cd hello-world-nodejs
-    $ docker-compose -f docker-compose.prod.yml up -d
+    $ cd nodejs-hello-world
+    $ DOCKER_IMAGE_REVIEW=hoatle/nodejs-hello-world:feature-1 \
+      docker-compose -f docker-compose.yml -f docker-compose.review.yml up -d review && \
+      docker-compose -f docker-compose.yml -f docker-compose.review.yml logs -f review
+
+After that, open:
+
+- http://review.nodejs.teracy.dev or https://review.nodejs.teracy.dev on your host browser to see
+  the app on the review mode.
+- Check out the VM's :ref:`basic_usage-ip_address` and on any device within your LAN,
+  open http://review.nodejs.<vm_ip>.xip.io or https://review.nodejs.<vm_ip>.xip.io to see the web app.
+
+
+Create CI/CD system
+-------------------
+
+It's required that we should always run CI/CD to automate the build and deployment. In this guide,
+we're going to create CI/CD for travis-ci and gitlab-ci.
+
+For deployment, we're going to deploy on Heroku, Google Container Engine (Kubernertes).
+
+The changes should be like this: https://github.com/acme101/nodejs-hello-world/commit/ca822a679691de619200e7cd2a0a5d946e5045ae
+
+More the most up to date and more information, please checkout the README.md file from the
```

Comment:
To have the most up-to-date  and more information, please checkout the ``README.md`` file....